src/include/common/type_utils.h

```diff
@@ -23,7 +23,14 @@ class StringCastUtils {
         if (iss.str().empty()) {
             throw ConversionException{"Empty string."};
         }
-        iss >> result;
+
+        if constexpr (std::is_same_v<int8_t, T>) {
```

Comment:
Why do you need a if statement here?  `iss >> result` should convert the string to an int8 value.

Reply:
Because >> treats int8_t as an unsigned char type in C++, it will result in iss.failed() if we don't do this.